ts/src/pro/poloniexfutures.ts

```diff
@@ -0,0 +1,741 @@
+            'id': this.safeString (trade, 'tradeId'),
+            'symbol': this.safeSymbol (marketId, market),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'order': this.safeString (trade, 'orderId'),
+            'type': this.safeStringLower (trade, 'type'),
+            'side': this.safeString (trade, 'side'),
+            'takerOrMaker': this.safeString (trade, 'matchRole'),
+            'price': this.safeString (trade, 'price'),
+            'amount': this.safeString (trade, 'tradeAmount'), // ? tradeQty?
+            'cost': undefined,
+            'fee': {
+                'rate': undefined,
+                'cost': this.safeString (trade, 'tradeFee'),
+                'currency': this.safeString (trade, 'feeCurrency'),
+            },
+        }, market);
+    }
+
+    handleOrder (client, message) {
+        //
+        //    {
+        //        data: {
+        //          symbol: 'ADAUSDTPERP',
+        //          orderType: 'limit',
+        //          side: 'buy',
+        //          canceledSize: '1',
+        //          orderId: '642b4d4c0494cd0007c76813',
+        //          type: 'canceled',
+        //          orderTime: '1680559436101909048',
+        //          size: '1',
+        //          filledSize: '0',
+        //          marginType: 1,
+        //          price: '0.25',
+        //          remainSize: '0',
+        //          clientOid: '112cbbf1-95a3-4917-957c-d3a87d81f853',
+        //          status: 'done',
+        //          ts: 1680559677560686600
+        //        },
+        //        subject: 'orderChange',
+        //        topic: '/contractMarket/tradeOrders',
+        //        channelType: 'private',
+        //        type: 'message',
+        //        userId: '1139790'
+        //    }
+        //
+        const data = this.safeValue (message, 'data');
+        let orders = this.orders;
+        if (orders === undefined) {
+            const limit = this.safeInteger (this.options, 'ordersLimit');
+            orders = new ArrayCacheBySymbolById (limit);
+            this.orders = orders;
+        }
+        const marketId = this.safeString (data, 'symbol');
+        if (marketId !== undefined) {
+            const messageHash = '/contractMarket/tradeOrders:' + marketId;
+            const symbol = this.safeSymbol (marketId);
+            const orderId = this.safeString (data, 'orderId');
+            const clientOrderId = this.safeString (data, 'clientOid');
+            const previousOrders = this.safeValue (orders.hashmap, symbol, {});
+            const previousOrder = this.safeValue2 (previousOrders, orderId, clientOrderId);
+            if (previousOrder === undefined) {
+                const parsed = this.parseWsOrder (data);
+                orders.append (parsed);
+                client.resolve (orders, messageHash);
+            } else {
+                const trade = this.parseWsTrade (data);
```

Comment:
If I understand right here you assuming if there was no previous order there is a trade? Would that cause issues when editting or canceling an order?

Reply:
I actually don't fully understand everything in this method, I copied the code from `coinbasepro` and modified it to fit this exchange

Reply:
In this case the function is looking if a previous order with the same id exists to help parse the order message. It makes sense in coinbasepro as they are different messages for different order events. In the case of Poloniex it looks like poloniex returns the full order information on each order message so you don't need to lookup the previous order if I'm reading it correctly

Reply:
How is poloniex returning the full order information? It looks like the trades linked to this order aren't provided, but since we're keeping track of all the orders, then when an update to an order is posted, we are able to deduce that there was a trade linked to this order, and add it to the trades array on our order object. 

I don't see any information about what trades are linked to this order provided by the exchange